src/discovery/dns/Akka.Discovery.Dns/Internal/AsyncDnsClient.cs

```diff
@@ -0,0 +1,421 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Net;
+using Akka.Actor;
+using Akka.Event;
+using Akka.IO;
+using Akka.Pattern;
+using Akka.Routing;
+
+namespace Akka.Discovery.Dns.Internal;
+
+/// <summary>
+/// DNS client actor for resolving DNS queries, including SRV records.
+/// This is an internal implementation for the Akka.Discovery.Dns service.
+/// </summary>
+internal class AsyncDnsClient(AsyncDnsCache cache, Configuration.Config config, EndPoint nameserver)
+    : UntypedActorWithStash
+{
+    #region Messages and internal types 
+    /// <summary>
+    /// Base class for DNS questions
+    /// </summary>
+    public record DnsQuestion(short Id, string Name, DnsProtocol.RecordType RecordType) : IConsistentHashable
+    {
+        public object ConsistentHashKey { get; } = Name; //not sure is this it the right approach 
+    }
+
+    /// <summary>
+    /// Request to drop a pending DNS question
+    /// </summary>
+    public sealed record DropRequest(DnsQuestion Question);
+
+    /// <summary>
+    /// Notification that a request has been dropped
+    /// </summary>
+    public sealed record Dropped(short Id);
+
+    /// <summary>
+    /// Message indicating TCP connection dropped
+    /// </summary>
+    public static readonly object TcpDropped = new object();
```

Comment:
This needs to be changed to an immutable singleton class instance and be handled by the actor. There are nothing in the code that handles this at the moment.